### ISSUE TITLE
Set position sometimes fails

### ISSUE THREAD

**Issue by mfld-fr** [skipperbent/happn-php-sdk#7]
Hello,

While testing a script to change periodically the position inside a region of interest, I have a strange behavior of the Happn API : it accepts the POST request the first time, then rejects the second request, and so on... whatever the delay between the requests (up to 15 minutes)...

The problem occurs with both the "rect bissect" and "square spiral" algorithms. I could have suspected a fake position detection on the first, but not on the second that moves the position like a real user (no huge jump in a step).

Any idea ?

**Comment by skipperbent** [comment]
Are you using the SDK to get the position? Because as far as I know the API has changed much since the SDK was initially created - and needs to be updated to work with the latest changes.

When I got some more time, i'll update it - but it's a huge step, as there is no documentation and because it's not officially supported. I think the best way to determine why the problem occurs would be to install the Android app and monitor the service-calls made by the app through some kind of proxy program.

**Comment by mfld-fr** [comment]
No, I am not using your SDK to **set** the position, but my own PHP code; this is more a question about the behaviour of the Happn REST API, not really related to this or that implementation.

And I found no clue yet by monitoring the Happn traffic with MITM sniffer on Android :-(

_Continuing the discussion here : https://github.com/rickhousley/happn/issues/9_

**Comment by skipperbent** [comment]
@mfld-fr 

How stable is the `rickhousley/happn-sdk`? Is it stable enough for me to port the changes to the PHP-SDK or would you recommend if I waited?

**Comment by mfld-fr** [comment]
Well, as far as I know, the Python part has been tuned recently by @SpringJools (fixed the User-Agent header as I did in your project), and the PHP part that I just submitted in the `php` branch works well these days : it is funny to let the algorithms exploring the area of interest all the day long and to browse the users they found with the tiny website :-)

The big limitation is the missing device creation, as we did not understand yet the device creation process (and so we have to copy the data caught in the sniffer), and the fact that sometimes the position setting fails, as reported above.

**Comment by skipperbent** [comment]
Yeah it's quite freaky how much you can learn about the users. Haha :) Along with their Facebook id :D Thanks for the reply, i'll look into your PHP code and get some inspiration for the missing methods etc. - great job by the way!

One thing that i noticed in my SDK is that it doesn't seem to update the position. When calling setposition, it report OK by the api - but the distance is always null when retrieving information about a user. Did you manage to fix that?

**Comment by mfld-fr** [comment]
I could not have done this funny experiment without the code base I found in the top-starred projects related to Happn hacking, including your 'SDK' ;-)

About the set position without actual effect : yeah, we got such behaviour, with HTTP 200 OK but unchanged latitude and longitude returned... it fortunately went away during the update of the code, so I am not able to explain it for now.

**Comment by springjools** [comment]
I have tested the position a little bit on myself, and I have noticed that the location sometimes gets "stuck" in some weird place for up to one day sometimes, like in a lake or under a bridge or some place where I know I wasn't. I thought a little what this could depend on, like maybe a different geodetic datum that is used in France or something (you know that there are different ways to output gps coordinates, but I do think that happn uses the same standard as google maps for instance, but it would be nice to have this verified).

The only way to get the position "unstuck" was to move significantly again. So i don't know if this helps but I wouldn't take the positions at face value really, I think they need to be confirmed to be valid with real location.
